casper/src/test/resources/PoSTest.rho

```diff
@@ -193,37 +193,39 @@ match (
                     @initialRewards <- initialRewardsCh) {
                   @PoS!("pay", 100, *retCh) |
                   for ( @(true, _) <- retCh) {
-
-                    @posVault!("balance", *finalPosBalanceCh) |
-                    @user1Vault!("balance", *finalUser1BalanceCh) |
-                    @PoS!("getRewards", *finalRewardsCh) |
-                    for(@finalPosBalance <- finalPosBalanceCh;
-                        @finalUser1Balance <- finalUser1BalanceCh;
-                        @finalRewards <- finalRewardsCh) {
-                      @ListOps!("fold", finalRewards.toList(), {}, *computeDelta, *deltaRewardsCh) |
-                      contract computeDelta(@(pk, finalReward), @acc, resultCh) = {
-                        resultCh!(acc.set(pk, finalReward - initialRewards.get(pk)))
-                      } |
-                      for (@deltaRewards <- deltaRewardsCh) {
-                        @ListOps!("forall", deltaRewards, *isPositive, *allRewardsPositiveCh) |
-                        contract isPositive(@n, resultCh) = { resultCh!(n > 0)} |
-
-                        @ListOps!("fold", deltaRewards.toList(), 0, *sumDelta, *deltaSumCh) |
-                        contract sumDelta(@(_, delta), @acc, resultCh) = {
-                          resultCh!(acc + delta)
+                    @PoS!("closeBlock", *ackCh) |
+                    for (_ <- ackCh) {
+                      @posVault!("balance", *finalPosBalanceCh) |
+                      @user1Vault!("balance", *finalUser1BalanceCh) |
+                      @PoS!("getRewards", *finalRewardsCh) |
```

Comment:
Will you add an assertion that the reward value of each validator in the `initialRewards` map is zero?

Reply:
> I cannot re-instantiate it for each test

It's not that you can't. You would just have to go through an entire test setup to do so (construct `RhoSpec`, `PoS`, `RevVault`, etc.) and that's cumbersome. There's a question of whether one `RhoSpec` instance should rerun the test setup for each test, or whether one `RhoSpec` instance should be created for each test.